### PR TITLE
SONARPY-1257 S1144: Should not raise on methods/classes with unknown decorator

### PR DESCRIPTION
**Pull request by maksim-grebeniuk-sonarsource**

### CODE REVIEW

**Review by guillaume-dequenne-sonarsource — CHANGES_REQUESTED** [review]
Looking good.
Left small comments comments.

File: python-checks/src/main/java/org/sonar/python/checks/AbstractUnreadPrivateMembersCheck.java [hunk 1]
```diff
@@ -19,30 +19,43 @@
...
+        .flatMap(Collection::stream)
         .filter(s -> s.name().startsWith(memberPrefix) && !s.name().endsWith("__") && equalsToKind(s) && isNeverRead(s))
-        .forEach(symbol -> reportIssue(ctx, symbol)));
+        .filter(this::filterMember)
+        .forEach(symbol -> reportIssue(ctx, symbol));
     });
   }
 
+  protected boolean filterMember(Symbol symbol) {
+    return true;
```
[the remark this answers is not in the document]
Comment (maksim-grebeniuk-sonarsource) [reply python-checks/src/main/java/org/sonar/python/checks/AbstractUnreadPrivateMembersCheck.java]: done

File: python-checks/src/main/java/org/sonar/python/checks/AbstractUnreadPrivateMembersCheck.java [hunk 2]
```diff
@@ -19,30 +19,43 @@
...
+        .stream()
+        .flatMap(Collection::stream)
         .filter(s -> s.name().startsWith(memberPrefix) && !s.name().endsWith("__") && equalsToKind(s) && isNeverRead(s))
-        .forEach(symbol -> reportIssue(ctx, symbol)));
+        .filter(this::filterMember)
+        .forEach(symbol -> reportIssue(ctx, symbol));
     });
   }
 
+  protected boolean filterMember(Symbol symbol) {
```
[the remark this answers is not in the document]
Comment (maksim-grebeniuk-sonarsource) [reply python-checks/src/main/java/org/sonar/python/checks/AbstractUnreadPrivateMembersCheck.java]: done, renamed to `isException`

File: python-frontend/src/main/java/org/sonar/python/semantic/BuiltinSymbols.java [hunk 3]
```diff
@@ -338,4 +338,9 @@ public static Set<String> all() {
     all.addAll(BuiltinSymbols.MODULE_ATTRIBUTES);
     return all;
   }
+
+  public static final String CLASS_METHOD_DECORATOR = "classmethod";
+  public static final String STATIC_METHOD_DECORATOR = "staticmethod";
+
+  public static final Set<String> KNOWN_DECORATORS = Set.of(CLASS_METHOD_DECORATOR, STATIC_METHOD_DECORATOR);
```
[the remark this answers is not in the document]
Comment (maksim-grebeniuk-sonarsource) [reply python-frontend/src/main/java/org/sonar/python/semantic/BuiltinSymbols.java]: done

**Review by guillaume-dequenne-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**